cfme/infrastructure/resource_pool.py

```diff
@@ -82,30 +77,53 @@ def get_detail(self, *ident):
             *ident: An InfoBlock title, followed by the Key name, e.g. "Relationships", "Images"
         Returns: A string representing the contents of the InfoBlock's value.
         """
-        if not self._on_detail_page():
-            sel.force_navigate('infrastructure_resource_pool', context=self._get_context())
+        navigate_to(self, 'Details')
         return details_page.infoblock.text(*ident)
 
-    def _on_detail_page(self):
-        """ Returns ``True`` if on the resource pool detail page, ``False`` if not."""
-        return sel.is_displayed(
-            '//div[@class="dhtmlxInfoBarLabel-2"][contains(., "{}") and contains(., "{}")]'.format(
-                self.name, "Summary")
-        )
-
     @property
     def exists(self):
         try:
-            sel.force_navigate('infrastructure_resource_pool', context=self._get_context())
-            quad = Quadicon(self.name, 'resource_pool')
+            navigate_to(self, 'All')
+            quad = Quadicon(self.name, self.quad_name)
             if sel.is_displayed(quad):
                 return True
         except sel.NoSuchElementException:
             return False
 
 
+@navigator.register(ResourcePool, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Compute', 'Infrastructure', 'Resource Pools')(None)
+
+    def resetter(self):
+        # Reset view and selection
+        tb.select("Grid View")
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
+
+
+@navigator.register(ResourcePool, 'Details')
+class Details(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        sel.click(Quadicon(self.obj.name, self.obj.quad_name))
+
+    def am_i_here(self):
```

Comment:
I've already come across other similar issues when I had to tie web object with its id.
Names are not unique. So, those cannot be used as ids.
From my perspective, we should tie web objects and ids and just check those ids. 
This is the only way I see how we can check that we use correct object (which leads to correct view as a result).
Concerning quad icons, those have href property which ends with object id. It also mentions the view it leads to. So, we can parse that href.